scripts/run_experiment.py

```diff
@@ -113,15 +113,25 @@ def run_experiment(argv):
 
     args = parser.parse_args(argv[1:])
 
-    assert (os.environ.get("JOBLIB_START_METHOD", None) == "forkserver")
     if args.seed is not None:
         set_seed(args.seed)
 
+    sigint_hdlr = signal.getsignal(signal.SIGINT)
+
+    def terminte_sampler(signum, frame):
+        parallel_sampler.terminate()
+        parallel_sampler.join()
```

Comment:
what if parallel_sampler hands after terminate? can we join() with a timeout and take more drastic termination steps if it hangs?

Reply:
Unfortunately the Pool class that MemmappingPool inherits from does not have a [join](https://docs.python.org/3.6/library/multiprocessing.html#multiprocessing.pool.Pool.join) method with timeout parameter.
From the Pool API, [close](https://docs.python.org/3.6/library/multiprocessing.html#multiprocessing.pool.Pool.close) is the nice way to ask to finish all tasks, and terminate just interrupts and finishes right away.

Reply:
Ok SGTM. We can test this and see how robust it is before going to greater lengths.

Reply:
Actually calling join here is wrong since the signal handler is called by the Python main thread and join has an assert call to make sure the parent process is trying to join.
Also, I found some handling that I did for this in a previous fix [here](https://github.com/rlworkgroup/garage/blob/a3f6fe4e2cc030b0b58d9dd403ba9a04c04dce7f/scripts/run_experiment.py#L174). I will add the join call there instead of the overriding the signal handler.